### PR TITLE
:zap: implement OnConfigurationChange to have direct change without r…

### LINKED ISSUE #22: Implement OnConfigurationChange of mattermost
Need to rebuild a gitlabClient when configuration change if organisation or privateEnable has change

### PR DESCRIPTION
**Pull request by manland**
…eload

Fix #22

### CODE REVIEW

File: server/configuration.go [hunk 1]
```diff
@@ -117,5 +121,34 @@ func (p *Plugin) OnConfigurationChange() error {
 
 	p.setConfiguration(configuration, serverConfiguration)
 
+	if err := configuration.IsValid(); err != nil {
+		return err
+	}
+
+	botID, ensureBotError := p.Helpers.EnsureBot(&model.Bot{
```
Comment (hanzei) [inline server/configuration.go]: Why do you want to do this on every config reload?

File: server/configuration.go [hunk 2]
```diff
@@ -117,5 +121,34 @@ func (p *Plugin) OnConfigurationChange() error {
...
+		Username:    "gitlab",
+		DisplayName: "GitLab Plugin",
+		Description: "A bot account created by the plugin GitLab.",
+	})
+	if ensureBotError != nil {
+		return errors.Wrap(ensureBotError, "can't ensure bot")
+	}
+	p.BotUserID = botID
+
+	p.WebhookHandler = webhook.NewWebhook(&gitlabRetreiver{p: p})
```
Comment (hanzei) [inline server/configuration.go]: Same question

File: server/configuration.go [hunk 3]
```diff
@@ -117,5 +121,34 @@ func (p *Plugin) OnConfigurationChange() error {
...
+	p.GitlabClient = gitlab.New(configuration.GitlabURL, configuration.GitlabGroup, p.checkGroup)
+
+	bundlePath, err := p.API.GetBundlePath()
```
Comment (hanzei) [inline server/configuration.go]: Same question.

**Comment by manland** [comment]
Yes @hanzei you right! I needed some break :face_with_thermometer: But now I'm here again :D 

Thanks to be my `gitkeeper`!

**Pull request merged**